hypothesis/lms
# Stop using LegacyCourse

For #3699 


After merging https://github.com/hypothesis/lms/pull/3808 and with the logging introduced in https://github.com/hypothesis/lms/pull/3788 we can check that we haven't found any occurrence where we would've used legacy course: https://my.papertrailapp.com/groups/14104911/events?q=Created+course+from+existing+legacy+course.+context_id since the back-fill was deployed.


This PR removes all code usage of LegacyCourse but leaves the table there in case we have to revert any of this.


## Testing

- Remove existing grouping/courses 


```
tox -qe dockercompose -- exec postgres psql -U postgres -c "truncate grouping,grouping_membership,course"
```


- Launch https://hypothesis.instructure.com/courses/125/assignments/873


- Check groupings has a row for the course

```
tox -qe dockercompose -- exec postgres psql -U postgres -c "select * from grouping where type = 'course'"
          created           |          updated           | id | application_instance_id |          authority_provided_id           | parent_id |                  lms_id                  |                  lms_name                   |  type  |                                                                                                                            settings                                                                                                                             |                     extra                      
----------------------------+----------------------------+----+-------------------------+------------------------------------------+-----------+------------------------------------------+---------------------------------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------
 2022-05-05 08:06:59.005543 | 2022-05-05 08:06:59.005543 |  8 |                       8 | 653e9620a656a954c684942f6443fa3e3410c03a |           | f3cd019017839c4630358662a05540f2f6ec5f93 | Developer Test Course with Sections Enabled | course | {"jstor": {"enabled": true, "site_code": null}, "canvas": {"groups_enabled": true, "sections_enabled": true}, "blackboard": {"files_enabled": false, "groups_enabled": false}, "vitalsource": {"enabled": true}, "microsoft_onedrive": {"files_enabled": true}} | {"canvas": {"custom_canvas_course_id": "125"}}
(1 row)
```

- Course remains empty

```
tox -qe dockercompose -- exec postgres psql -U postgres -c "select * from course"                        
 consumer_key | authority_provided_id | settings 
--------------+-----------------------+----------
(0 rows)

```


- Relaunch https://hypothesis.instructure.com/courses/125/assignments/873
- The same grouping row is re-used.
